applications/IgaApplication/custom_elements/embedded_isogeometric_beam_element.cpp

```diff
@@ -0,0 +1,5629 @@
+                                            - sn * _phi_der * phi_var[r] * permutation[t][k][u] * _vec_var[k * N_Dof + s]
+                                                - sn * _phi_der * phi_var[s] * permutation[t][k][u] * _vec_var[k * N_Dof + r]
+                                                + cs * _phi_der * permutation[t][k][u] * _vec_var_var(k * N_Dof + r, s)
+                                                - phi_var[r] * phi_var[s] * sn * permutation[t][k][u] * _vec_der[k]
+                                                + phi_var[r] * cs * permutation[t][k][u] * _vec_der_var[k * N_Dof + s]
+                                                    + phi_var[s] * cs * permutation[t][k][u] * _vec_der_var[k * N_Dof + r]
+                                                    ;
+
+                                                //else      
+                                                _mat_rod_der_var_var(t * N_Dof + r, u * N_Dof + s) += sn * permutation[t][k][u] * _vec_der_var_var(k * N_Dof + r, s); //*_mat_identity(u,u)
+
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    void EmbeddedIsogeometricBeamElement::comp_mat_rodrigues_deriv2(Matrix3d& _mat_rod_derder, Vector3d _vec, Vector3d _vec_deriv, Vector3d _vec_deriv2, float _phi, float _phi_deriv, float _phi_deriv2)
+    {
+        _mat_rod_derder.clear();
+
+        Matrix3d _mat_identity;
+        _mat_identity.clear();  //initialization by 0 
+        for (int i = 0; i < 3; i++) { _mat_identity(i, i) = 1; }
+
+        for (int i = 0; i < 3; i++) { _mat_rod_derder(i, i) = -_phi_deriv2 * sin(_phi) - pow(_phi_deriv, 2) * cos(_phi); }
+        _mat_rod_derder += cross_prod_vec_mat(_vec, _mat_identity) * (cos(_phi) * _phi_deriv2 - pow(_phi_deriv, 2) * sin(_phi));
+        _mat_rod_derder += cross_prod_vec_mat(_vec_deriv, _mat_identity) * 2 * _phi_deriv * cos(_phi);
+        _mat_rod_derder += cross_prod_vec_mat(_vec_deriv2, _mat_identity) * sin(_phi);
+    }
+
+    void EmbeddedIsogeometricBeamElement::comp_mat_rodrigues_deriv2_var(Matrix& _mat_rod_derder_var, Vector3d _vec, Vector _vec_var, Vector3d _vec_der, Vector _vec_der_var, Vector3d _vec_derder, Vector _vec_derder_var, Vector _func, Vector _deriv, Vector _deriv2, float _phi, float _phi_der, float _phi_der2)
+    {
+        //const unsigned int N_Dof = 4;//this->GetGeometry().PointsNumber()* (this->GetGeometry().WorkingSpaceDimension() + 1);
+        _mat_rod_derder_var.resize(3 * N_Dof, 3);
+        _mat_rod_derder_var.clear();
+
+        int permutation[3][3][3];
+        for (int i = 0; i < 3; i++)
+        {
+            for (int j = 0; j < 3; j++)
+            {
+                for (int k = 0; k < 3; k++)
+                {
+                    permutation[i][j][k] = 0;
+                }
+            }
+        }
+
+        permutation[0][1][2] = 1;
```

Comment:
this code is repeated many times, besides we can simplify the 3rd order tensor i think